### PR TITLE
Firefox: Add Data Collection Consent Step in Initialization

### LINKED ISSUE #1198: [Demand] Firefox Market Compliance: Data Collection Consent Page in Initialization
## Metadata

Field | Value
--- | ---
Previous Stage | ...
Figma Link | Figma_Link_Here____
Expected UX prototype due | 2020-W??-5
Actual UX prototype release | -
Expected engineering due | 2020-W??-5
Corresponding Pull Request | -
Completion Version | -

## Introduction

We will add a Data Collection Consent page in Initialization, per the market policy of Firefox market.

> “I noticed you're collecting information from the user's facebook account. You need to create a consent page (aka control mechanism for data collection), which is shown at install time. The consent page should contain a choice accompanied by the data collection summary. The choice to send data shouldn't be enabled by default. Here is an example of how to create a page within the add-on https://blog.mozilla.org/addons/2016/07/15/writing-an-opt-in-ui-for-an-extension/. Also, the summary of the data collection should be mentioned in the add-on description and privacy policy. For more information, refer to https://extensionworkshop.com/documentation/publish/add-on-policies/#Data_Disclosure_Collection_and_Management .“

## Place

This page should be added as the initial welcome page before inputting the name.

## Content

- Title: Welcome to Maskbook
- Subtitle: Make Privacy Protected Again
- Body:
  - Be aware of your data when using Maskbook.
  - Maskbook will collect some data when you use, but the developers of Maskbook have no access to them. The collected data are stored locally, except for the data which are necessary for enabling your friends to decrypt the posts you shared.
  - The data being collected include: Your social network profile URL, the Maskbook public keys of your contacts (friends, who you follow, and who follow you), the relations between you and your contacts, the cleartext of your posts, the encryption keys of your posts, and the data managed by each Plugin in Maskbook (they have their own privacy policies).
  - [Checkbox] I understand this and accept the [privacy policy](https://maskbook.com/privacy-policy/).
  - Button: Get Started

## Notes

- None.

### PR DESCRIPTION
**Pull request by guanbinrui**
maskbook-data-collection-consent-page

Note: This page pops up only one-time after installed the extension and can be easily skipped by change URL in the location.

close #1198

### CODE REVIEW

File: src/extension/options-page/Welcome/getWelcomePageURL.tsx [hunk 1]
```diff
@@ -13,6 +13,6 @@ export function getWelcomePageURL(query?: Query) {
         if (identifier) param.set('identifier', identifier.toText())
         return getUrl(`index.html#/home/?${param.toString()}`)
     } else {
-        return getUrl(`index.html#/initialize/`)
+        return getUrl(`index.html#/initialize${webpackEnv.target === 'Firefox' ? '/1d' : ''}`)
```
Comment (septs) [inline src/extension/options-page/Welcome/getWelcomePageURL.tsx:16]:
```suggestion
        if (webpackEnv.target === 'Firefox') {
            return t('initialize_firefox_url')
        } else {
            return t('initialize_general_url')
        }
```

reference: #816

Comment (neruthes) [reply src/extension/options-page/Welcome/getWelcomePageURL.tsx:16]: lgtm

**Comment by Jack-Works** [comment]
Users might think we are dishonest on the privacy if we only display this on Firefox.

File: src/extension/options-page/InitStep.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 export enum InitStep {
-    'Setup1' = '1s',
-    'Setup2' = '2s',
-    'Restore1' = '1r',
-    'RestoreAdvanced1' = '1ra',
-    'Restore2' = '2r',
+    Disclaimer = '1d',
```
Comment (Jack-Works) [inline src/extension/options-page/InitStep.ts:2]: the number "0" means it is the 0th step. the after char "s" or "r" is mean the different branch of the `n`th step.

If you think the old naming logic is not descriptive, feel free to move the naming style to `"step-0-disclaimer"`

File: src/extension/options-page/DashboardInitSteps/Disclaimer.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+export default function Disclaimer() {
+    const { t } = useI18N()
+    const [checked, setChecked] = useState(false)
+    const header = 'Be aware of your data when using Maskbook.'
+    const content = (
+        <div style={{ width: '100%' }}>
+            <LinedBox>
+                <Typography>
+                    Maskbook will collect some data (listed in the privacy policy) when you use, but the developers of
```
Comment (Jack-Works) [inline src/extension/options-page/DashboardInitSteps/Disclaimer.tsx:29]: It should be `i18n`ed

File: src/extension/options-page/Welcome/getWelcomePageURL.tsx (hunk 1 above)
Comment (guanbinrui) [reply src/extension/options-page/Welcome/getWelcomePageURL.tsx:16]: This URL should be generated programmatically. I think it's not necessary to move URL fragments to I18N settings

File: src/extension/options-page/InitStep.ts (hunk 2 above)
Comment (guanbinrui) [reply src/extension/options-page/InitStep.ts:2]: Temporarily naming it as `-1s` and the old initialization implementation will be replaced totally in [feature/dashboard](https://github.com/DimensionDev/Maskbook/pull/1026).

File: src/_locales/zh/messages.json [hunk 4]
```diff
@@ -61,6 +61,11 @@
     "dashboard_import_persona_hint": "導入以前的角色備份。",
     "dashboard_imported_persona": "導入了角色 `{{name}}`，包含 {{count}} 個資料.",
     "dashboard_init_header": "在社交網絡上發布信息，但不允許他們跟踪您。",
+    "dashboard_init_step_-1": "使用 Maskbook 時請注意您的數據。",
+    "dashboard_init_step_-1_hint": "Maskbook 在使用時會收集一些數據（在隱私政策中列出），但是 Maskbook 的開發人員無法訪問它們。 除了使您的朋友解密您共享的帖子所需的數據外，收集的數據存儲在本地。",
+    "dashboard_init_step_-1_privacy_prefix": "我同意",
+    "dashboard_init_step_-1_privacy": "隱私政策",
+    "dashboard_init_step_-1_privacy_suffix": "。",
```
Comment (Jack-Works) [inline src/_locales/zh/messages.json:68]: 🤣🤣

**Comment by Jack-Works** [comment]
> Users might think we are dishonest on the privacy if we only display this on Firefox.

@neruthes

**Comment by guanbinrui** [comment]
> > Users might think we are dishonest on the privacy if we only display this on Firefox.
> 
> @neruthes

Let's display it for everybody!

**Comment by neruthes** [comment]
> > > Users might think we are dishonest on the privacy if we only display this on Firefox.
> 
> > 
> 
> > @neruthes
> 
> 
> 
> Let's display it for everybody!

And this is actually what I personally preferred at the beginning. Now we have a majority consensus. Let it be universal. 🎉

**Pull request merged**